cmd/blockchaincmd/import_public.go

```diff
@@ -221,62 +180,103 @@ func importPublic(*cobra.Command, []string) error {
+
+	createChainTx, err := utils.GetBlockchainTx(network.Endpoint, blockchainID)
 	if err != nil {
-		return models.Sidecar{}, err
+		return models.Sidecar{}, nil, err
 	}
 
+	subnetID := createChainTx.SubnetID
+	vmID := createChainTx.VMID
+	blockchainName := createChainTx.ChainName
+	genBytes := createChainTx.GenesisData
+
+	printFunc("Retrieved information:")
+	printFunc("  Name: %s", blockchainName)
+	printFunc("  BlockchainID: %s", blockchainID.String())
+	printFunc("  SubnetID: %s", subnetID.String())
+	printFunc("  VMID: %s", vmID.String())
+
 	subnetInfo, err := blockchain.GetSubnet(subnetID, network)
 	if err != nil {
-		return models.Sidecar{}, err
-	}
-	if subnetInfo.IsPermissioned {
-		return models.Sidecar{}, fmt.Errorf("unable to import non sovereign Subnets")
+		return models.Sidecar{}, nil, err
 	}
-	validatorManagerAddress = "0x" + hex.EncodeToString(subnetInfo.ManagerAddress)
 
-	// add validator without blockchain arg is only for l1s
-	sc = models.Sidecar{
-		Sovereign: true,
+	sc := models.Sidecar{
+		Name: blockchainName,
+		Networks: map[string]models.NetworkData{
+			network.Name(): {
+				SubnetID:     subnetID,
+				BlockchainID: blockchainID,
+			},
+		},
+		Subnet:          blockchainName,
+		Version:         constants.SidecarVersion,
+		ImportedVMID:    vmID.String(),
+		ImportedFromAPM: true,
 	}
 
-	sc.ValidatorManagement, err = validatorManagerSDK.GetValidatorManagerType(rpcURL, common.HexToAddress(validatorManagerAddress))
-	if err != nil {
-		return models.Sidecar{}, fmt.Errorf("could not obtain validator manager type: %w", err)
+	if rpcIsAvailable {
+		e := sc.Networks[network.Name()]
+		e.RPCEndpoints = []string{rpcURL}
+		sc.Networks[network.Name()] = e
 	}
 
-	if sc.ValidatorManagement == validatormanagertypes.ProofOfAuthority {
-		owner, err := contract.GetContractOwner(rpcURL, common.HexToAddress(validatorManagerAddress))
-		if err != nil {
-			return models.Sidecar{}, err
+	if !subnetInfo.IsPermissioned {
+		sc.Sovereign = true
+		validatorManagerAddress = "0x" + hex.EncodeToString(subnetInfo.ManagerAddress)
+		e := sc.Networks[network.Name()]
+		e.ValidatorManagerAddress = validatorManagerAddress
+		sc.Networks[network.Name()] = e
+		printFunc("  Validator Manager Address: %s", validatorManagerAddress)
+		if rpcIsAvailable {
+			sc.ValidatorManagement, err = validatorManagerSDK.GetValidatorManagerType(rpcURL, common.HexToAddress(validatorManagerAddress))
+			if err != nil {
+				return models.Sidecar{}, nil, fmt.Errorf("could not obtain validator manager type: %w", err)
+			}
+			printFunc("  Validation Kind: %s", sc.ValidatorManagement)
+			if sc.ValidatorManagement == validatormanagertypes.ProofOfAuthority {
+				owner, err := contract.GetContractOwner(rpcURL, common.HexToAddress(validatorManagerAddress))
+				if err != nil {
+					return models.Sidecar{}, nil, err
+				}
+				sc.ValidatorManagerOwner = owner.String()
+				printFunc("  Validator Manager Owner: %s", sc.ValidatorManagerOwner)
```

Comment:
nit: move this new block to its own member func

Reply:
can you clarify what this new function abstracts